drivebrain_core_base/src/Configurable.cpp

```diff
@@ -0,0 +1,50 @@
+#include <Configurable.hpp>
+
+using namespace core::common;
+std::string Configurable::get_name()
+{
+    return _component_name;
+}
+
+std::vector<std::string> Configurable::get_param_names()
+{
+
+    std::vector<std::string> names;
+    {
+        std::unique_lock lk(_live_params.mtx);
```

Comment:
is there no need to unlock in these places?

Reply:
when the lock goes out of scope the lock unlocks / gets deleted.

https://stackoverflow.com/questions/51683326/is-using-unique-lock-in-new-scope-equivalent-to-unlock-call-at-the-end-of-work-w